### PR TITLE
MNT-20764: Searches fail for users who are members of groups where the authorityName contains double quotes

### PR DESCRIPTION
**Pull request by amohammedalfresco**
- Appended double quote (`"`) character  to invalid string array in main code and test check

### CODE REVIEW

**Comment by killerboot** [comment]
Is there a plan to fix it for other APIs like CMIS?
The changes probably need to happen on repository level. I suspect that would be more appropriate and would fix the REST API endpoint, too.

**Comment by killerboot** [comment]
@amohammedalfresco If the fix in https://github.com/Alfresco/alfresco-repository/pull/933 is applied.
Does the test here pass without changes in GroupsImpl ?

**Comment by amohammedalfresco** [comment]
> @amohammedalfresco If the fix in [Alfresco/alfresco-repository#933](https://github.com/Alfresco/alfresco-repository/pull/933) is applied.
> Does the test here pass without changes in GroupsImpl ?

I used  our BAT environment with changes including `alfresco-repository` but it seems the creation of groups in remote-api is defined exclusively in the project, and it is not related to `alfresco-repository` method of createAuthority.  Hence, the `GroupsImpl` needed adjustment here.

**Comment by killerboot** [comment]
Ideally the changes in alfresco-repository should be enough to protect the REST API endpoint form creating Users and Groups with bad characters. Please investigated the way authorities are created via REST APIs and make necessary changes deeper in backend. In this case not only V1 REST API will be protected, but other endpoints as well, including custom code in extensions.

File: src/main/resources/alfresco/public-rest-context.xml [hunk 1]
```diff
@@ -140,6 +140,7 @@
                 <entry key="org.alfresco.rest.framework.core.exceptions.ApiException" value="#{T(org.springframework.extensions.webscripts.Status).STATUS_INTERNAL_SERVER_ERROR}" />
                 <entry key="java.lang.Throwable" value="#{T(org.springframework.extensions.webscripts.Status).STATUS_INTERNAL_SERVER_ERROR}" />
                 <entry key="java.lang.IllegalArgumentException" value="#{T(org.springframework.extensions.webscripts.Status).STATUS_BAD_REQUEST}" />
+                <entry key="org.alfresco.error.AlfrescoRuntimeException" value="#{T(org.springframework.extensions.webscripts.Status).STATUS_BAD_REQUEST}" />
```
Comment (killerboot) [inline src/main/resources/alfresco/public-rest-context.xml:143]: AlfrescoRuntimeExceptions are valid 500 errors. Unfortunately this change can't be done.

Comment (amohammedalfresco) [reply src/main/resources/alfresco/public-rest-context.xml]: Sure.  Renamed then accordingly to their module exception class.

**Pull request merged**